playground/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useCallback, useDeferredValue } from "react";
+import CustomTest from "./CustomTest.tsx";
+import ExcalidrawWrapper from "./ExcalidrawWrapper.tsx";
+import Testcases from "./Testcases.tsx";
+import { parseMermaid } from "../src/parseMermaid.ts";
+
+export interface MermaidData {
+  definition: string;
+  output: Awaited<ReturnType<typeof parseMermaid>> | null;
+  error: string | null;
+}
+
+const App = () => {
+  const [mermaidData, setMermaidData] = useState<MermaidData>({
+    definition: "",
+    output: null,
+    error: null,
+  });
+
+  const mermaidDefinition = useDeferredValue(mermaidData.definition);
+
+  const handleUpdateSyntax = useCallback(
+    async (definition: MermaidData["definition"]) => {
+      setMermaidData({
+        definition,
+        output: null,
+        error: null,
+      });
+    },
+    []
+  );
+
+  const handleDataParsed = useCallback(
+    (parsedData: MermaidData["output"], err?: unknown) => {
+      setMermaidData({
+        definition: mermaidData.definition,
+        output: parsedData,
+        error: err ? String(err) : null,
+      });
+    },
+    [mermaidData]
+  );
```

Comment:
This is slightly confusing, can we combine this to one callback instead and use it for both?
Ideally, the function signature should be the same for both.

Reply:
nice catch! done in [192f306](https://github.com/excalidraw/mermaid-to-excalidraw/pull/52/commits/192f30684dcb2c81bffe4dfb80d0b99b7e9399e3)